### PR TITLE
4.x: fix deprecation notice related to Chronos 2.4

### LINKED ISSUE #17223: 2.5 timezone() is deprecated. Use setTimezone() instead
### Description

`i18nFormat` causes this error due to the following dependency
```
"cakephp/chronos": "^2.2"
```

It is getting chronos 2.4.0 when I compose install, which deprecates the timezone function

### CakePHP Version

4.4

### PHP Version

_No response_

### PR DESCRIPTION
**Pull request by LordSimal**
resolves https://github.com/cakephp/cakephp/issues/17223

### CODE REVIEW

**Comment by LordSimal** [comment]
Should we just ignore those stan errors? Most of these would be resolved if we directly do
```
class FrozenDate extends \Cake\Chronos\ChronosDate
```
but we can't since `\Cake\Chronos\ChronosDate` only exists in Chronos 2.4+

Or should we adjust our min. chronos version to `^2.4`? Would this be doable in a patch release?

**Comment by ravage84** [comment]
> Or should we adjust our min. chronos version to ^2.4? Would this be doable in a patch release?

Even though we could, it would force everybody to use the new Chronos minor going forward, which could lead to issues for some.

> Should we just ignore those stan errors?

Thus, yes, I would for 4.4.

File: src/I18n/DateFormatTrait.php [hunk 1]
```diff
@@ -190,7 +190,7 @@ public function i18nFormat($format = null, $timezone = null, $locale = null)
         if ($timezone) {
             // Handle the immutable and mutable object cases.
             $time = clone $this;
-            $time = $time->timezone($timezone);
+            $time = $time->setTimezone($timezone);
```
Comment (ADmad) [inline src/I18n/DateFormatTrait.php:193]: If we change the name then the constraints for chronos should be updated to the minor in which this method was added (assuming setTimezone() doesn't exist in 2.2).

Comment (LordSimal) [reply src/I18n/DateFormatTrait.php:193]: `setTimezone()` does exist in [Chronos 2.2](https://github.com/cakephp/chronos/blob/2.2.0/src/Traits/TimezoneTrait.php#L55)

Comment (ADmad) [reply src/I18n/DateFormatTrait.php:193]: Great, so no other change needed.

**Comment by othercorey** [comment]
> Should we just ignore those stan errors? Most of these would be resolved if we directly do

We'll have to ignore some of them since we now extend deprecated classes.

**Comment by othercorey** [comment]
The setTimezone() change looks good.

**Review by othercorey — APPROVED** [review]

**Pull request merged**